src/main/kotlin/nl/info/zac/authentication/UserPrincipalFilter.kt

```diff
@@ -76,9 +89,67 @@ constructor(
                     setLoggedInUserOnHttpSession(userPrincipal as OidcPrincipal<*>, httpSession)
                 }
         }
+
+        if (!isAuthorizationAllowed(request)) {
+            response.sendError(jakarta.servlet.http.HttpServletResponse.SC_FORBIDDEN)
+            return
+        }
+
         filterChain.doFilter(servletRequest, servletResponse)
     }
 
+    @Suppress("ReturnCount")
+    private fun isAuthorizationAllowed(request: HttpServletRequest): Boolean {
```

Comment:
I'm not clear about the name of this method. I read it per below, but correct me if I'm wrong.

To me the name suggests that this is a check to see if authorization is allowed on this request. A next step will be the actual authorization itself. Maybe a better name might then be
```suggestion
    private fun needsRequestAuthorization(request: HttpServletRequest): Boolean {
```
or
```suggestion
    private fun isAuthorizationEnabledForRequest(request: HttpServletRequest): Boolean {
```
or maybe you have a better idea?